core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -130,9 +130,17 @@ public static FileSystem create(ClientContext ctx) {
      * @return a new FileSystem instance
      */
     public static FileSystem create(FileSystemContext context) {
+      return create(context, context.getClusterConf());
```

Comment:
Would it be cleaner if we just have a subclass of `FileSystemContext` that override `getClusterConf` to return the client side configuration instead?

Reply:
`FileSystemContext` wasn't written to be easily extended, and even if we had a subclass we would still need a switch somewhere to make that FileSystemContext version